plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -286,4 +287,46 @@ public static void saveLibraryList(IJavaProject project, List<Library> libraries
     }
   }
 
+  /**
+   * Check that the library list is still required (i.e., the classpath container may have been
+   * removed).
+   */
+  public static void checkLibraryList(IJavaProject javaProject, IProgressMonitor monitor) {
+    SubMonitor progress = SubMonitor.convert(monitor, 5);
+    LibraryClasspathContainerSerializer serializer = new LibraryClasspathContainerSerializer();
+    if (!serializer.hasLibraryIds(javaProject)) {
+      // nothing to do since no library list file found
+      return;
+    }
+    progress.worked(1);
+
+    // Check if we can find our master library container
+    IClasspathEntry[] rawClasspath;
```

Comment:
might be clearer if we used one try block and pushed this declaration and following into it